### PR TITLE
edit update functions implemented

### PR DESCRIPTION
**Pull request by axnish27**
Implemented @Raza9798  #22

### CODE REVIEW

File: resources/views/admin.blade.php [hunk 1]
```diff
@@ -108,18 +134,47 @@ function deleteEmployee(id){
...
+                let $id = null
+
+                //Edit Employee
+                $('#myTable tbody').on('click', 'button[id^="btn-edit"]', function (e) {
+                    const $data =  table.row( $(this).parents('tr') ).data();
+                    $id  = $data.id
+                    
+                    $('#modal-edit').each(function (){
+                        const $inputs =  $(this).find(':input');
+                        $($inputs[1]).val($data.name);
```
Comment (Raza9798) [inline resources/views/admin.blade.php:158]: better to use class name or element id when you set values

Comment (axnish27) [reply resources/views/admin.blade.php]: changes made please review again @Raza9798 #22

File: resources/views/admin.blade.php [hunk 2]
```diff
@@ -146,22 +146,19 @@ function deleteEmployee(id){
                     });
                 });
 
-                let $id = null
 
                 //Edit Employee
+                let $id = null
                 $('#myTable tbody').on('click', 'button[id^="btn-edit"]', function (e) {
```
Comment (Raza9798) [inline resources/views/admin.blade.php:152]: access the button using `ID` instead of `element`

File: resources/views/admin.blade.php [hunk 3]
```diff
@@ -146,22 +146,19 @@ function deleteEmployee(id){
...
                     const $data =  table.row( $(this).parents('tr') ).data();
                     $id  = $data.id
```
Comment (Raza9798) [inline resources/views/admin.blade.php:154]: this naming conventions `$id` and `$data` will make you trouble once you deep dive into the large code base. just skip this comment now. i will explain once you come to that stage

**Review by Raza9798 — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/EmployeeController.php [hunk 4]
```diff
@@ -53,7 +53,17 @@ public function store(Request $request){
     }
 
     public function update(Request $request , string $id){
-        return('admin');
+
+        Employee::find($id)->update([
+            'name'=>$request->name,
+            'position'=>$request->position,
+            'dob'=>$request->dob,
```
Comment (Raza9798) [inline app/Http/Controllers/EmployeeController.php:60]: I saw that you have made the migration with `date` type for column `dob` what will happened if the request send with other values except date format

**Review by Raza9798 — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/EmployeeController.php (hunk 4 above)
Comment (axnish27) [reply app/Http/Controllers/EmployeeController.php:60]: image

throws error.....form not submitted #22 @Raza9798 

shall i add backend validations?

File: resources/views/admin.blade.php (hunk 3 above)
Comment (axnish27) [reply resources/views/admin.blade.php:154]: Okay 👍

File: app/Http/Controllers/EmployeeController.php (hunk 4 above)
Comment (Raza9798) [reply app/Http/Controllers/EmployeeController.php:60]: yes please add it @axnish27

Comment (axnish27) [reply app/Http/Controllers/EmployeeController.php:60]: okay sure @Raza9798

Comment (axnish27) [reply app/Http/Controllers/EmployeeController.php]: Validation for both edit and create functions implemented please review @Raza9798 #22

File: resources/views/admin.blade.php (hunk 2 above)
Comment (axnish27) [reply resources/views/admin.blade.php:153]: okay noted

Comment (axnish27) [reply resources/views/admin.blade.php]: resolved please check @Raza9798  #22

File: resources/views/admin.blade.php [hunk 5]
```diff
@@ -117,7 +117,7 @@
                 });
 
                 // Delete Employee
-                $('#myTable tbody').on('click', 'button[id^="btn-dlt"]', function (e) {
+                $('#myTable tbody').on('click', '[id^="btn-dlt"]', function (e) {
```
Comment (Raza9798) [inline resources/views/admin.blade.php:120]: use `id` selector instead of `attribute` selector

**Review by Raza9798 — CHANGES_REQUESTED** [review]

File: resources/views/admin.blade.php (hunk 5 above)
Comment (axnish27) [reply resources/views/admin.blade.php]: okay got it.....used class selector and made the change please review @Raza9798  #22

**Pull request merged**